waku/v2/protocol/peer_exchange/shard_lru.go

```diff
@@ -0,0 +1,137 @@
+package peer_exchange
+
+import (
+	"container/list"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/waku-org/go-waku/waku/v2/protocol"
+	wenr "github.com/waku-org/go-waku/waku/v2/protocol/enr"
+)
+
+type ClusterIndex struct {
+	cluster uint16
+	index   uint16
+}
+type shardLRU struct {
```

Comment:
Rather than implementing an LRU ourselves, doesn't it make sense to use an existing LRU implementation to store nodes per shard?

Reply:
Got it. 
My point is that if we can avoid spending time on implementing DS that are available and tested, as such code would be difficult to maintain.